### DISCUSSION TITLE
Javascript Customizations

### DISCUSSION THREAD

**Discussion by Erin-Cecele** [thegetty/quire#427] [Development]
As we continue to improve Quire, we'd love to learn more about what kind of javascript you included in your projects (custom, existing libraries, etc.) that wasn't already part of Quire and how/where you added it.

@yl5682 @daniel-keller

**Comment by daniel-keller** [comment]
We only added a few JS libraries for the Clowes Catalog project. 

* I decided to go with OpenseaDragon for our deepzoom instead of leaflet as we had some more complex zoom widgets (side-by-side and slider comparisons) and it was what I'm familiar with. 
* Instead of PrincePDF I generated PDFs through a headless chrome instance using [chrome-headless-render-pdf](https://www.npmjs.com/package/chrome-headless-render-pdf).
* Hugo has built in code minification but `quire site|build` wasn't using it (that I could find at least). I wrote a script using [html-minifier](https://www.npmjs.com/package/html-minifier) to minify the final build. Looking at the quire `package.json` just now the `build:production` command does minify but I'm not seeing that in the Quire documentation.

We're working on building out a React component library for our branding so I could see us integrate this library in future publications.

**Reply by Erin-Cecele** [reply]
Thank you, @daniel-keller for this very helpful feedback. We will bring this to the attention of our software engineers.
